api/api/handlers/product.py

```diff
@@ -76,20 +85,65 @@ def save_product(request):
     """ handles both update and create requests """
     data = request.json_body
     try:
-        product = request.context.get_product(request.subpath[0])
-    except InvalidId:
-        return {"message": "Invalid id given"}
-    except DoesNotExist:
+        validate(data, json.loads(request.context.product_schema))
+    except ValidationError:
+        return {"message": "invalid data"}
+    if request.subpath:
+        try:
```

Comment:
This should actually become 2 seperate methods. "POST" and "PUT" that both call a `product_save` method while passing a product. For now ignore but we kiiiinda have to refactor that and it actually will make our lives easier.